src/Hintman/Core/Hint.hs

```diff
@@ -10,6 +15,27 @@ module Hintman.Core.Hint
 import Relude.Extra.Enum (inverseMap)
 
 
+-- | Data type to represent the information for comment creation.
+data Hint = Hint
+    { hintHeader       :: !Text  -- ^ Hint text
+    , hintBody         :: !Text  -- ^ New line body
+    , hintIsSuggestion :: !Bool  -- ^ Is this GitHub @suggestion@?
+    , hintNote         :: !Text  -- ^ Additional information
+    , hintType         :: !HintType  -- ^ Type of the hint
+    } deriving stock (Show, Eq)
+
+-- | Formats 'Hint' in the compatible with @GitHub@ way.
+formatHint :: Hint -> Text
+formatHint Hint{..} = unlines $ concat
+    [ [ hintHeader | hintHeader /= ""]
+    , ["```" <> if hintIsSuggestion then "suggestion" else ""]
+    , [ hintBody | hintType /= TrailingNewlines ]
+    , [ "```"]
+    , [ "__Note:__ " <> hintNote | hintNote /= "" ]
+    ]
```

Comment:
Wow, this is actually very cool to write this function and take all cases into consideration :open_mouth: :boom: